Improve documentation of API kinds (#6341)
CVC4/CVC4: docs/cpp/kind.rst

Kind
====

Every :cpp:class:`Term <cvc5::api::Term>` has a kind which represents its type, for example whether it is an equality (:cpp:enumerator:`EQUAL <cvc5::api::Kind::EQUAL>`), a conjunction (:cpp:enumerator:`AND <cvc5::api::Kind::AND>`), or a bitvector addition (:cpp:enumerator:`BITVECTOR_PLUS <cvc5::api::Kind::BITVECTOR_PLUS>`).

> Note that the api type :cpp:enum:`cvc5::api::Kind` roughly corresponds to :cpp:enum:`cvc5::Kind`, but is a different type and has a few subtle differences.

Reviewer: ```suggestion
Note that the API type :cpp:enum:`cvc5::api::Kind` roughly corresponds to :cpp:enum:`cvc5::Kind`, but is a different type and has a few subtle differences.
```

Reviewer: I'm not sure if we should even mention that. `cvc5::Kind` should not be accessible from outside at all.

Kind
====

Every :cpp:class:`Term <cvc5::api::Term>` has a kind which represents its type,
for example whether it is an equality (:cpp:enumerator:`EQUAL
<cvc5::api::Kind::EQUAL>`), a conjunction (:cpp:enumerator:`AND
<cvc5::api::Kind::AND>`), or a bitvector addition
(:cpp:enumerator:`BITVECTOR_PLUS <cvc5::api::Kind::BITVECTOR_PLUS>`).

Note that the API type :cpp:enum:`cvc5::api::Kind` roughly corresponds to

> :cpp:enum:`cvc5::Kind`, but is a different type and has a few subtle

Reviewer: ```suggestion
:cpp:enum:`cvc5::Kind`, but is a different type. It hides internal kinds that
should not be exported to the API, and maps all kinds that we want to export
to its corresponding internal kinds.
```

> differences.

Reviewer: ```suggestion
```
